x-pack/legacy/plugins/siem/public/pages/case/components/configure_cases/field_mapping_row.tsx

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState, useCallback, useMemo } from 'react';
+import { EuiFlexItem, EuiFlexGroup, EuiSuperSelect, EuiIcon } from '@elastic/eui';
+
+import * as i18n from './translations';
+
+interface ThirdPartyField {
+  value: string;
+  label: string;
+}
+interface RowProps {
+  siemField: string;
+  thirdPartyFields: ThirdPartyField[];
+}
+
+const editUpdateOptions = [
+  {
+    value: 'nothing',
+    inputDisplay: <span>{i18n.FIELD_MAPPING_EDIT_NOTHING}</span>,
+    'data-test-subj': 'edit-update-option-nothing',
+  },
+  {
+    value: 'overwrite',
+    inputDisplay: <span>{i18n.FIELD_MAPPING_EDIT_OVERWRITE}</span>,
+    'data-test-subj': 'edit-update-option-overwrite',
+  },
+  {
+    value: 'append',
+    inputDisplay: <span>{i18n.FIELD_MAPPING_EDIT_APPEND}</span>,
+    'data-test-subj': 'edit-update-option-append',
+  },
+];
+
+const FieldMappingRowComponent: React.FC<RowProps> = ({ siemField, thirdPartyFields }) => {
+  const [selectedEditUpdate, selectEditUpdate] = useState(editUpdateOptions[0].value);
+  const onEditUpdateChange = useCallback(option => selectEditUpdate(option), [selectedEditUpdate]);
+
+  const thirdPartyOptions = useMemo(
+    () =>
+      thirdPartyFields.map(field => ({
+        value: field.value,
+        inputDisplay: <span>{field.label}</span>,
+      })),
+    [thirdPartyFields]
+  );
+
+  const [selectedThirdParty, setSelectedThirdParty] = useState(thirdPartyOptions[0].value);
```

Comment:
could you move state declaration to the top of the component?

Reply:
I need `thirdPartyOptions` for this state declaration. Should I move the `thirdPartyOptions` declaration at the top before the declaration of all states?

Reply:
maybe we should do that in `FieldMappingComponent` because in current state we will do the mapping for each `FieldMappingRow`